data-model/src/main/java/io/micronaut/data/model/runtime/RuntimePersistentEntity.java

```diff
@@ -306,6 +306,15 @@ public RuntimePersistentProperty<T> getIdentity() {
         return identity.length == 1 ? identity[0] : null;
     }
 
+    @Override
+    public List<PersistentProperty> getIdentityProperties() {
+        return List.of(identity);
+    }
+
+    public List<RuntimePersistentProperty<T>> getRuntimeIdentityProperties() {
```

Comment:
Is this duplicate/not needed?

Reply:
It's returning runtime properties, unfortunately the original method cannot have a different type